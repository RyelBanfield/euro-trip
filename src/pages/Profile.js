import { useSelector } from 'react-redux';
import CountryComponent from '../components/Country';

const Profile = () => {
  const user = useSelector((state) => state.user);
  const countries = useSelector((state) => state.countries);
  const favourites = useSelector((state) => state.favourites);

  const currentUserFavourites = favourites.filter((fav) => fav.user_id === user.id)
    .map((fav) => fav.country_id);
  const currentUserFavouriteCountries = countries.filter((country) => currentUserFavourites
    .includes(country.id));

  return (
    <main className="profile fade-in">
      <img className="profile-picture" src={user.picture} alt={user.name} />
      <h2 className="profile-name">{user.name}</h2>
      <p className="profile-email">{user.email}</p>
      <div className="favourites">
        <h3>Favourites</h3>
        {currentUserFavouriteCountries.length > 0 && (
        <div className="countries">
          {currentUserFavouriteCountries.map((country) => (
            <CountryComponent
              country={country}
              isFavourited={
                user
                && user.countries
                  ? user.countries
                    .find((favouriteCountry) => favouriteCountry.name === country.name)
                    !== undefined
                  : false
              }
              key={country.id}
            />
          ))}
        </div>
        )}
      </div>
    </main>
  );
};

export default Profile;
