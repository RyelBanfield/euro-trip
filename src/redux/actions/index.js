const LOAD_COUNTRIES_INTO_STORE = 'LOAD_COUNTRIES_INTO_STORE';

// eslint-disable-next-line import/prefer-default-export
export const loadCountriesIntoStore = (countries) => ({
  type: LOAD_COUNTRIES_INTO_STORE,
  payload: countries,
});
