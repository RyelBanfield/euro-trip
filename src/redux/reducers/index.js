import { combineReducers } from 'redux';
import userReducer from './user';
import countriesReducer from './countries';
import imagesReducer from './images';
import favouritesReducer from './favourites';

const rootReducer = combineReducers({
  user: userReducer,
  countries: countriesReducer,
  images: imagesReducer,
  favourites: favouritesReducer,
});

export default rootReducer;
