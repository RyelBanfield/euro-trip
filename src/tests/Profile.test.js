import { render, screen } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import rootReducer from '../redux/reducers';
import Profile from '../pages/Profile';

const store = createStore(rootReducer);

it('renders correctly', () => {
  const profile = renderer
    .create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    )
    .toJSON();
  expect(profile).toMatchSnapshot();
});

it('renders main', () => {
  render(
    <Provider store={store}>
      <Profile />
    </Provider>,
  );
  const main = screen.getByRole('main');
  expect(main).toBeInTheDocument();
});

it('renders heading', () => {
  render(
    <Provider store={store}>
      <Profile />
    </Provider>,
  );
  const heading = screen.getByRole('heading');
  expect(heading).toBeInTheDocument();
});
