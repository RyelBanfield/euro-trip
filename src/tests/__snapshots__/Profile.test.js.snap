// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="profile fade-in"
>
  <div
    className="favourites"
  >
    <h3>
      Favourites
    </h3>
  </div>
</main>
`;
